src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -234,17 +357,73 @@ namespace winrt::TerminalApp::implementation
         _noMatchesText().Visibility(_filteredActions.Size() > 0 ? Visibility::Collapsed : Visibility::Visible);
     }
 
-    Collections::IObservableVector<Command> CommandPalette::FilteredActions()
+    Collections::IObservableVector<TerminalApp::Command> CommandPalette::FilteredActions()
     {
         return _filteredActions;
     }
 
-    void CommandPalette::SetActions(Collections::IVector<TerminalApp::Command> const& actions)
+    void CommandPalette::SetCommands(Collections::IVector<TerminalApp::Command> const& actions)
     {
-        _allActions = actions;
+        _allCommands = actions;
         _updateFilteredActions();
     }
 
+    void CommandPalette::EnableCommandPaletteMode()
+    {
+        _switchToMode(CommandPaletteMode::ActionMode);
+        _updateFilteredActions();
+    }
+
+    void CommandPalette::_switchToMode(CommandPaletteMode mode)
+    {
+        // The smooth remove/add animations that happen during
+        // UpdateFilteredActions don't work very well when switching between
+        // modes because of the sheer amount of remove/adds. So, let's just
+        // clear + append when switching between modes.
+        if (mode != _currentMode)
+        {
+            _currentMode = mode;
+            _filteredActions.Clear();
+            auto commandsToFilter = _commandsToFilter();
+
+            for (auto action : commandsToFilter)
+            {
+                _filteredActions.Append(action);
+            }
+        }
+
+        switch (_currentMode)
+        {
+        case CommandPaletteMode::ActionMode:
+            SearchBoxText(RS_(L"CommandPalette_SearchBox/PlaceholderText"));
+            NoMatchesText(RS_(L"CommandPalette_NoMatchesText/Text"));
+            ControlName(L"Command Palette");
+            break;
+        case CommandPaletteMode::TabSwitcherMode:
+        {
+            SearchBoxText(RS_(L"TabSwitcher_SearchBoxText"));
+            NoMatchesText(RS_(L"TabSwitcher_NoMatchesText"));
+
+            if (_anchorKey != VirtualKey::None)
+            {
+                _searchBox().Visibility(Visibility::Collapsed);
+                ControlName(L"Tab Switcher Anchored Mode");
+            }
+            else
+            {
+                ControlName(L"Tab Switcher UnAnchored Mode");
```

Comment:
These may also want to be localized.. but perhaps the name doesn't need to differ between anchored/unanchored :smile: